sdks/python/apache_beam/transforms/window.py

```diff
@@ -341,14 +343,18 @@ def __ne__(self, other):
 
   def to_runner_api_parameter(self, context):
     return (urns.FIXED_WINDOWS_FN,
-            proto_utils.pack_Struct(size=self.size.micros,
-                                    offset=self.offset.micros))
-
-  @urns.RunnerApiFn.register_urn(urns.FIXED_WINDOWS_FN, struct_pb2.Struct)
+            standard_window_fns_pb2.FixedWindowsPayload(
+                size=proto_utils.from_micros(
+                    duration_pb2.Duration, self.size.micros),
+                offset=proto_utils.from_micros(
+                    timestamp_pb2.Timestamp, self.offset.micros)))
+
+  @urns.RunnerApiFn.register_urn(
```

Comment:
There are certainly pros and cons to each, but we should be consistent. Certainly going from the urn to object requires some kind of registration (even if the other way around can be handled by polymorphism). 

Probably wouldn't hurt to put together a doc detailing these design choices, which wouldn't have to all be resolved now (but before the Runner/Fn Api is public we would make sure all SDKs are consistent).